fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -576,12 +580,18 @@ public long fileLength(@Nonnull String name) throws NoSuchFileException {
      */
     @Override
     @Nonnull
+    @SuppressWarnings("java:S2093")
     public IndexOutput createOutput(@Nonnull final String name, @Nullable final IOContext ioContext) {
         if (LOGGER.isTraceEnabled()) {
             LOGGER.trace(getLogMessage("createOutput",
                     LuceneLogMessageKeys.FILE_NAME, name));
         }
-        return new FDBIndexOutput(name, name, this);
+        long startTime = System.nanoTime();
+        try {
+            return new FDBIndexOutput(name, name, this);
+        } finally {
+            context.record(LuceneEvents.Waits.WAIT_LUCENE_CREATE_OUTPUT, System.nanoTime() - startTime);
```

Comment:
Why are you adding tracking for this? I don't immediately see anything that should be expensive

Reply:
There are some potentially meaningful things to capture in `fdbDirectory.getIncrement()`which is called from the constructor. Plus, incrementing the counter (even without the WAIT) would give us meaningful data of the flow of file creation.